craft_cli/dispatcher.py

```diff
@@ -334,7 +334,9 @@ def _build_no_command_error(self, missing_command: str) -> str:
         msg = f"no such command {missing_command!r}{extra_similar}"
         return self._help_builder.get_usage_message(msg)
 
-    def _parse_options(self, defined_arguments, sysargs):  # pylint: disable=too-many-branches
+    def _parse_options(
```

Comment:
We might want to take the type checker related changes and put them in their own PR. (As long as CI still succeeds)

Reply:
Hmmm yeah good point, there's no great way to do this. Retracted, let's keep it this way since it's less work and not particularly unclear.